category/category.go

```diff
@@ -0,0 +1,271 @@
+			"category_id": category.ID,
+			"err":         db.Error.Error(),
+		}, "unable to create category")
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	log.Info(ctx, map[string]interface{}{
+		"category_id": category.ID,
+	}, "category created successfully")
+	return category, nil
+}
+
+// LoadAllRelationshipsOfCategory loads all the relationships of a category. This is required for workitemtype filtering.
+func (m *GormRepository) LoadAllRelationshipsOfCategory(ctx context.Context, categoryID uuid.UUID) ([]*WorkItemTypeCategoryRelationship, error) {
+	// Check if category is present
+	_, err := m.LoadCategoryFromDB(ctx, categoryID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+		}, "category not found")
+		return nil, errors.NewBadParameterError("categoryID", categoryID)
+	}
+
+	relationship := []*WorkItemTypeCategoryRelationship{}
+	db := m.db.Model(&relationship).Where("category_id=?", categoryID).Find(&relationship)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+		}, "workitemtypes of category not found")
+		return nil, errors.NewNotFoundError("work item type category", categoryID.String())
+	}
+	if err := db.Error; err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+			"err":         err,
+		}, "unable to list workitemtype category relationships")
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	return relationship, nil
+}
+
+// LoadCategoryFromDB returns category for the given id
+// This is needed to check if a category is present in db or not.
+func (m *GormRepository) LoadCategoryFromDB(ctx context.Context, id uuid.UUID) (*Category, error) {
+	res := Category{}
+	db := m.db.Model(&res).Where("id=?", id).First(&res)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": id,
+		}, "category not found")
+		return nil, errors.NewNotFoundError("category", id.String())
+	}
+	if err := db.Error; err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": id,
+			"err":         err,
+		}, "unable to load category", err.Error())
+		return nil, errors.NewInternalError(err.Error())
+	}
+	return &res, nil
+}
+
+// LoadWorkItemTypeCategoryRelationship loads all the relationships of a category. This is required for testing.
+func (m *GormRepository) LoadWorkItemTypeCategoryRelationship(ctx context.Context, workitemtypeID uuid.UUID, categoryID uuid.UUID) (*WorkItemTypeCategoryRelationship, error) {
+	// Check if category is present
+	_, err := m.LoadCategoryFromDB(ctx, categoryID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+		}, "category not found")
+		return nil, errors.NewBadParameterError("categoryID", categoryID)
+	}
+	relationship := WorkItemTypeCategoryRelationship{}
+	db := m.db.Model(&relationship).Where("category_id=? AND workitemtype_id=?", categoryID, workitemtypeID).Find(&relationship)
```

Comment:
@DhritiShikhar this `workitemtype_id` won't work after your change in the sql table